nuxeo-services/nuxeo-platform-directory/nuxeo-platform-directory-mongodb/src/main/java/org/nuxeo/directory/mongodb/MongoDBSession.java

```diff
@@ -487,7 +489,7 @@ public DocumentModelList query(QueryBuilder queryBuilder, boolean fetchReference
             long count;
             if (countTotal) {
                 // we have to do an additional query to count the total number of results
-                count = getCollection().count(filter);
+                count = countDocuments(DIRECTORY_CONNECTION_PREFIX + getDirectory().getName(), getCollection(), filter);
```

Comment:
In `MongoDBDirectory.initialize` could you just store the `databaseId` in a field and just refer to that everywhere it's needed instead of recomputing please?